sites/fast-color-explorer/app/components/swatch.ts

```diff
@@ -0,0 +1,200 @@
+import { contrastRatio, parseColor } from "@microsoft/fast-colors";
+import { attr, css, customElement, html, observable } from "@microsoft/fast-element";
+import { fillColor, neutralForegroundHint, Swatch } from "@microsoft/fast-components";
+import { elevation } from "@microsoft/fast-components/dist/esm/styles/elevation.js";
+import { DesignToken, FoundationElement } from "@microsoft/fast-foundation";
+
+export enum SwatchTypes {
+    fill = "fill",
+    foreground = "foreground",
+    outline = "outline",
+}
+
+const template = html<AppSwatch>`
+    <template class="${x => x.type}">
+        <div
+            class="icon"
+            style="${x => x.iconStyle}"
+            title="${x => x.contrastMessage}"
+        ></div>
+        <code class="recipe-name">
+            ${x => x.recipeName}
+        </code>
+        <code class="hex-code">
+            ${x => x.colorValue}
+        </code>
+    </template>
+`;
+
+const styles = css`
+    :host {
+        display: grid;
+        grid-template-columns: auto 1fr auto;
+        grid-template-rows: auto;
+        align-items: center;
+        width: 100%;
+        padding: 4px 0;
+        box-sizing: border-box;
+        color: ${neutralForegroundHint};
+        font-size: 12px;
+        grid-column-gap: 16px;
+        justify-items: start;
+    }
+    :host(.fill) {
+    }
+    :host(.foreground) .icon {
+        border: 1px solid black;
+    }
```

Comment:
Just a note that if these are still rendering into React here *is* a chance that the classes could be clobbered. I wonder if we should tie these to the attributes if they exist...

Reply:
These are only rendeded within another web component. Does that still apply if they are not direct children of a React component?

Reply:
You should be fine - I mentioned it primarily because I wasn't sure if this was rendering directly in React or in another WC

Reply:
Good to know. The plan is to get rid of all React entirely. Soon. :)